tasks.py

```diff
@@ -133,3 +273,96 @@ def install_local(ctx, package):
             pip(ctx, f"uninstall {pkg} -y")
             with ctx.cd(valid_packages[pkg]):
                 poetry(ctx, "install")
+
+
+@task
+def install_node(ctx):
+    """Installs and configures a node instance in the poetry .venv.
+    Primarily used for ``Playwright`` tasks.
+    """
+    poetry(ctx, "run rfbrowser init --skip-browsers")
+
+
+@task(pre=[install])
+def build_libdocs(ctx):
+    """Generates library specification and documentation using ``docgen``"""
+    libspec_promise = docgen(
+        ctx,
+        "rpaframework",
+        "--no-patches",
+        "--format libspec",
+        "--output docs/source/libspec/",
+        *DOCGEN_EXCLUDES,
+        asynchronous=True,
+    )
+    html_promise = docgen(
+        ctx,
+        "rpaframework",
+        "--template docs/source/template/libdoc/libdoc.html",
+        "--format html",
+        "--output docs/source/include/libdoc/",
+        *DOCGEN_EXCLUDES,
+        asynchronous=True,
+    )
+    json_promise = docgen(
+        ctx,
+        "rpaframework",
+        "--no-patches",
+        "--format json-html",
+        "--output docs/source/json/",
+        *DOCGEN_EXCLUDES,
+        asynchronous=True,
+    )
+    libspec_promise.join()
+    html_promise.join()
+    json_promise.join()
+    shutil.copy2(
+        "docs/source/template/iframeResizer.contentWindow.map",
+        "docs/source/include/libdoc/",
+    )
+
+
+@task(pre=[install, build_libdocs])
+def build_docs(ctx):
+    """Builds documentation locally. These can then be browsed directly
+    by going to ./docs/build/html/index.html or using ``invoke local-docs``
+    and navigating to localhost:8000 in your browser.
+
+    If you are developing documentation for an optional package, you must
+    use the appropriate ``invoke install-local`` command first.
+    """
+    sphinx(ctx, f"-M clean {DOCS_SOURCE_DIR} {DOCS_BUILD_DIR}")
+    python_tool(ctx, "todos", "packages/main/src", "docs/source/contributing/todos.rst")
+    python_tool(
+        ctx,
+        "merge",
+        "docs/source/json/",
+        "docs/source/include/latest.json",
+    )
+    sphinx(ctx, f"-b html -j auto {DOCS_SOURCE_DIR} {DOCS_BUILD_DIR}")
+    python_tool(ctx, "rss")
+
+
+@task
+def local_docs(ctx):
+    """Hosts library documentation on a local http server. Navigate to
+    localhost:8000 to browse."""
+    poetry(ctx, "run python -m http.server -d docs/build/html/")
+
+
+@task
+def install_hooks(ctx):
+    """Installs standard git hooks."""
+    git(ctx, f"config core.hooksPath {GIT_HOOKS_DIR}")
+
+
+@task
+def changelog(ctx):
+    """Prints changes in latest release."""
+    python_tool(ctx, "changelog")
+
+
+@task
+def lint_each(ctx):
+    """Executes linting on all packages."""
+    invoke_each(ctx, "lint")
```

Comment:
This task could also use parallelization to improve performance I feel like. So I could add error handling to even confirm it can initiate a lint and then initiate them all simultaneously and retain the results and print to `stdout` once they are all recovered.

Reply:
By using parallelization and setting it to warn on failure, it will run the invoke command for each package and report errors.